### PR TITLE
 [PlanDefinitionList] setState on unmounted component

### LINKED ISSUE #336: Fix Warning: An update to PlanDefinitionList was not wrapped in act
```console.error node_modules/react-dom/cjs/react-dom.development.js:506
      Warning: An update to PlanDefinitionList inside a test was not wrapped in act(...).
      
      When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
          in PlanDefinitionList (at tests/index.test.tsx:37)
          in Router (created by WrapperComponent)
          in WrapperComponent
    console.error node_modules/react-dom/cjs/react-dom.development.js:506
      Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
          in PlanDefinitionList (at tests/index.test.tsx:37)
          in Router (created by WrapperComponent)
          in WrapperComponent
```

### LINKED ISSUE #356: Fix UpdatePlan bug
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in PlanDefinitionList (created by Context.Consumer)
    in Connect(PlanDefinitionList) (created by Route)
```

### PR DESCRIPTION
**Pull request by p-netm**
- force async code to resolve before component unmounts in test.
- Track the loading status as a computed variable of the plans prop 

closes #356 
closes #336

### CODE REVIEW

**Comment by p-netm** [comment]
> Have you tested that this solves it? Is there a way to add a test for this?

I will send you a link for Pr once, i feel its ready, tests and all.

File: src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx [hunk 1]
```diff
@@ -33,6 +33,9 @@ const PlanDefinitionList = (props: PlanListProps) => {
   const { fetchPlans, plans, service } = props;
   const [loading, setLoading] = useState<boolean>(true);
 
+  /** Track status of the component  during a single render */
+  let isMounted = true;
+
   const apiService = new service('/plans');
```
Comment (p-netm) [inline src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx]: @moshthepitt  should this be `plans` or `/plans`?

Comment (moshthepitt) [reply src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx]: it should be `plans`.  Does a test exist that checks either we call the service correctly, or that the URL mock call is as expected? @p-netm

Comment (p-netm) [reply src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx]: yeap, will make sure there are there

**Comment by moshthepitt** [comment]
@p-netm   what is the status of this?  Do you think you'll be able to get this closed, or should we close this PR?

**Comment by p-netm** [comment]
@moshthepitt  This one is awaiting rereview and hopefully a merge.

**Comment by moshthepitt** [comment]
@p-netm can you fix the conflicts?

File: src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx [hunk 2]
```diff
@@ -39,7 +39,6 @@ interface PlanListProps {
 /** Simple component that loads the new plan form and allows you to create a new plan */
 const PlanDefinitionList = (props: PlanListProps) => {
   const { fetchPlans, plans, service } = props;
-  const [loading, setLoading] = useState<boolean>(true);
```
Comment (moshthepitt) [inline src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx]: @p-netm do you need to create an issue for all other components that used this approach?

Comment (p-netm) [reply src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx]: #554

**Comment by moshthepitt** [comment]
@p-netm please fix conflicts

**Comment by moshthepitt** [comment]
@p-netm is this superceded by #625?

**Comment by p-netm** [comment]
> @p-netm is this superceded by #625?

yes

**Pull request closed without merging**